app/controllers/distributions_controller.rb

```diff
@@ -43,30 +43,33 @@ def index
 
     @distributions = current_organization
                      .distributions
-                     .includes(:partner, :storage_location, line_items: [:item])
-                     .order('issued_at DESC')
-                     .apply_filters(filter_params.except(:date_range), helpers.selected_range)
+                     .order(issued_at: :desc)
+                     .includes(:partner, :storage_location)
+                     .class_filter(scope_filters)
     @paginated_distributions = @distributions.page(params[:page])
-    @items = current_organization.items.alphabetized
-    @item_categories = current_organization.item_categories
-    @storage_locations = current_organization.storage_locations.active_locations.alphabetized
-    @partners = @distributions.collect(&:partner).uniq.sort_by(&:name)
+    @items = current_organization.items.alphabetized.select(:id, :name)
+    @item_categories = current_organization.item_categories.select(:id, :name)
+    @storage_locations = current_organization.storage_locations.active_locations.alphabetized.select(:id, :name)
+    @partners = Partner.joins(:distributions).where(distributions: @distributions).distinct.order(:name).select(:id, :name)
```

Comment:
We just opened up https://github.com/rubyforgood/human-essentials/issues/4834 because the partners here don't really make sense to be limited to the other filters, btw. No need to change it in this PR but thought I'd mention it :)

Reply:
Thanks for the heads up. It would be a quick fix (either in this PR or separately). But since I've already attached a performance refactor to a bug fix in this PR, I'm wary to add more to it as it could lengthen the review process.

Reply:
Ah I see that ticket has high priority, so maybe I will just go ahead and create a separate PR and fix that.

Reply:
Created that PR. If it gets merged before this, I will merge it and fix any conflicts.

Reply:
(nods) High priority is a relative thing -- it's high priority in the context of we have 80-mumble issues and we don't want it to get lost, but not emergency.